src/generate.ts

```diff
@@ -269,16 +281,45 @@ class Generator {
   private getTypeFromPropertiesSchemaObj(
     obj: OpenAPIV3.NonArraySchemaObject
   ): string {
+    const requiredProps = this.getRequiredProperties(obj)
     const fields = Object.entries(obj.properties).map(([name, value]) => {
       const readOnly = (value as any).readOnly
-      const required = (obj.required || []).includes(name)
+      const description = (value as any).description
+      const required = requiredProps.includes(name)
 
-      return formatTypeField(readOnly, name, required, this.getType(value))
+      let retVal = ""
+      if (this.withDoc && description) {
+        retVal = `/** ${description} */\n  `
+      }
+      return (
+        retVal + formatTypeField(readOnly, name, required, this.getType(value))
+      )
```

Comment:
let's keep string manipulation consistent (don't mix interpolation and concatenation)

```suggestion
      return `${retVal}${formatTypeField(readOnly, name, required, this.getType(value))}`
```